File: dayTwo/Python/book_suggestion_system_Test.py
from unittest import TestCase

import book_suggestion_system

class AddBookMethod(TestCase):

    def test_that_add_book_function_exists(self):
        book = "gift of men"
        book_suggestion_system.add_book(book)
    
    def test_that_add_book_function_returns_added_succesfully(self):
        book = "gift of men"
        expected = "Book added succesfully"
        actual = book_suggestion_system.add_book(book)
        self.assertEqual(expected,actual)
       
 
class RemoveBookMethod(TestCase):

    def test_that_remove_book_function_exists(self):
        book = "gift of men"
        book_suggestion_system.remove_book(book)
    
    def test_that_remove_book_function_removes_specified_book_succesfull(self):
        book = "gift of men"
        expected = "Book removed successfully"
        actual = book_suggestion_system.remove_book(book)
        self.assertEqual(expected,actual)


class UpdateBookMethod(TestCase):
    book = "Greatest"
    book_suggestion_system.add_book(book)

    def test_that_update_book_function_exists(self):
        book_to_update = "gift of men"
        update = "light"
        book_suggestion_system.update_book(book_to_update,update)
    
    def test_that_update_book_function_updates_book_succesfully(self):

        book_to_update = "Greatest"
        update = "light"
        expected = "Book updated successfully"
        actual = book_suggestion_system.update_book(book_to_update,update)
        self.assertEqual(expected,actual)
        

class ShowBookMethod(TestCase):
    
    def test_that_show_book_function_exists(self):
        book_suggestion_system.show_books()
    
    def test_that_show_book_function_returns_all_books_in_library(self):
        expected =  ["Gift of men","King of persia","Lord of the rings","Super bowl","Greatest"]
        actual = book_suggestion_system.show_books()
        self.assertEqual(expected,actual)

        
#class SuggestBookMethod(TestCase):
#    
#    def test_that_suggest_book_function_exists(self):
#        book_suggestion_system.suggest_book()
#    
#    def test_that_suggest_book_function_returns_book_and_page_number(self):
#        expected = "Greatest page 0"
#        actual = book_suggestion_system.suggest_book()
#        self.assertEqual(expected,actual)
#        
#

File: dayTwo/javascript/bookSuggestionSystem.test.js
//importing all functions from bookSuggestionSystem
const {showBooks,addBook,removeBook,updateBook,getSuggestions} = require("./bookSuggestionSystem.js");

//Show books test
test("test that show books function returns all books", ()=>{
    
    expect(showBooks()).toEqual(["Divine","Gift of men"]);    

})

//add book test
test("test that add book function adds book successfully", ()=>{
    let bookToAdd = "lions den";
    let expected = "Book added successfully";
    
    expect(addBook(bookToAdd)).toEqual(expected);    

})



//remove book test
test("test that remove book function remove book successfully", ()=>{

    let bookToRemove = "lions den";
    let expected = "Book removed successfully";
    expect(removeBook(bookToRemove)).toEqual(expected);    

})


//update book test
test("test that update book function update book successfully", ()=>{
    let bookToAdd = "Game of thrones"    
    addBook(bookToAdd)
    let bookToUpdate = "Game of thrones";
    let update = "Game show";
    let expected = "Book updated successfully";
    expect(updateBook(bookToUpdate,update)).toEqual(expected);    

})


















//get suggestions test
//test("test that getSuggestions book function update book successfully", ()=>{
//    
//    let expected = "Divine page 0";
//    expect(getSuggestions()).toEqual(expected);    
//
//})
//


